spec/unit/ssl/state_machine_spec.rb

```diff
@@ -1049,5 +1069,40 @@ def write_csr_attributes(data)
         expect(state.next_state).to be_an_instance_of(Puppet::SSL::StateMachine::LockFailure)
       end
     end
+
+    context 'in state NeedRenewedCert' do
+      before :each do
+        client_cert.not_after=(Time.now + 300)
+      end
+
+      let(:ssl_context) { Puppet::SSL::SSLContext.new(cacerts: cacerts, client_cert: client_cert, crls: crls,)}
+      let(:state) { Puppet::SSL::StateMachine::NeedRenewedCert.new(machine, ssl_context, private_key) }
+      let(:renewed_cert) { cert_fixture('signed-renewed.pem') }
+
+      it 'returns Done with renewed cert when successful' do
+        allow(cert_provider).to receive(:save_client_cert)
+        stub_request(:post, %r{puppet-ca/v1/certificate_renewal}).to_return(status: 200, body: client_cert.to_pem)
+
+        st = state.next_state
+        expect(st).to be_an_instance_of(Puppet::SSL::StateMachine::Done)
+        # expect(st.ssl_context[:client_cert]).to eq(renewed_cert)
+      end
+
+      it 'logs an error message when failing with a non-404 status' do
+        allow(cert_provider).to receive(:save_client_cert)
```

Comment:
```suggestion
```
I'd drop this since we shouldn't attempt to save the client cert if the renewal fails.